pkg/vault/test/foundry/fork/BoostedPoolQueryVsActualOperation.t.sol

```diff
@@ -0,0 +1,491 @@
+        amountIn = bound(amountIn, 2 * DAI_BUFFER_INIT_AMOUNT, 4 * DAI_BUFFER_INIT_AMOUNT);
+        IBatchRouter.SwapPathExactAmountIn[] memory paths = _buildExactInPaths(daiFork, amountIn, 0);
+
+        uint256 snapshotId = vm.snapshot();
+        _prankStaticCall();
+        (uint256[] memory queryPathAmountsOut, , ) = batchRouter.querySwapExactIn(paths, bytes(""));
+        vm.revertTo(snapshotId);
+
+        vm.prank(lp);
+        (uint256[] memory actualPathAmountsOut, , ) = batchRouter.swapExactIn(paths, MAX_UINT256, false, bytes(""));
+
+        assertEq(queryPathAmountsOut[0], actualPathAmountsOut[0], "Query and actual outputs do not match");
+    }
+
+    function testDaiInOutOfBufferExactOut__Fork__Fuzz(uint256 amountOut) public {
+        amountOut = bound(amountOut, 2 * USDC_BUFFER_INIT_AMOUNT, 4 * USDC_BUFFER_INIT_AMOUNT);
+        IBatchRouter.SwapPathExactAmountOut[] memory paths = _buildExactOutPaths(
+            daiFork,
+            (2 * amountOut * DAI_FACTOR) / USDC_FACTOR,
+            amountOut
+        );
+
+        uint256 snapshotId = vm.snapshot();
+        _prankStaticCall();
+        (uint256[] memory queryPathAmountsIn, , ) = batchRouter.querySwapExactOut(paths, bytes(""));
+        vm.revertTo(snapshotId);
+
+        vm.prank(lp);
+        (uint256[] memory actualPathAmountsIn, , ) = batchRouter.swapExactOut(paths, MAX_UINT256, false, bytes(""));
+
+        assertEq(queryPathAmountsIn[0], actualPathAmountsIn[0], "Query and actual outputs do not match");
+    }
+
+    function testDaiInBufferUnbalancedExactIn__Fork__Fuzz(
+        uint256 amountIn,
+        uint256 unbalancedDai,
+        uint256 unbalancedUsdc
+    ) public {
+        unbalancedDai = bound(unbalancedDai, 0, DAI_BUFFER_INIT_AMOUNT);
+        unbalancedUsdc = bound(unbalancedUsdc, 0, USDC_BUFFER_INIT_AMOUNT);
+        _unbalanceBuffers(unbalancedDai, unbalancedUsdc);
+
+        amountIn = bound(amountIn, 2 * DAI_BUFFER_INIT_AMOUNT, 4 * DAI_BUFFER_INIT_AMOUNT);
+        IBatchRouter.SwapPathExactAmountIn[] memory paths = _buildExactInPaths(daiFork, amountIn, 0);
+
+        uint256 snapshotId = vm.snapshot();
+        _prankStaticCall();
+        (uint256[] memory queryPathAmountsOut, , ) = batchRouter.querySwapExactIn(paths, bytes(""));
+        vm.revertTo(snapshotId);
+
+        vm.prank(lp);
+        (uint256[] memory actualPathAmountsOut, , ) = batchRouter.swapExactIn(paths, MAX_UINT256, false, bytes(""));
+
+        assertEq(queryPathAmountsOut[0], actualPathAmountsOut[0], "Query and actual outputs do not match");
+    }
+
+    function testDaiInBufferUnbalancedExactOut__Fork__Fuzz(
+        uint256 amountOut,
+        uint256 unbalancedDai,
+        uint256 unbalancedUsdc
+    ) public {
+        unbalancedDai = bound(unbalancedDai, 0, DAI_BUFFER_INIT_AMOUNT);
+        unbalancedUsdc = bound(unbalancedUsdc, 0, USDC_BUFFER_INIT_AMOUNT);
+        _unbalanceBuffers(unbalancedDai, unbalancedUsdc);
```

Comment:
Question: in these cases we're just using the DAI buffer; do we need USDC? Is it just to have a single `_unbalanceBuffers` function?

Reply:
maybe it's fine and a refactor would be too much work for little gain, but maybe just fuzz a single amount and zero out the one we're not using?

Reply:
Nope, we're using both. The batch swap has 3 steps: Buffer -> Boosted Pool -> Buffer

In this case, is `Buffer DAI -> Boosted Pool -> Buffer USDC` ("DAI-first" scenario).

So, the balance of both buffers alters the operation.

I tested the `Buffer USDC -> Boosted Pool -> Buffer DAI` ("USDC-first" scenario) scenario too, because DAI has 18 decimals and USDC has 6, which can create rounding issues not covered by the "DAI-first" scenario.